plugins/signals-backend/src/service/SignalManager.ts

```diff
@@ -62,10 +63,18 @@ export class SignalManager {
   }
 
   private constructor(options: SignalManagerOptions) {
+    this.events = options.events;
     ({ events: this.events, logger: this.logger } = options);
 
+    // Use a unique subscriber ID for each signals instance, in order to fan-out
+    // all events to each signals instance. This ensures that events always
+    // reach users in a scaled deployment.
+    const id = `signals-${crypto.randomBytes(8).toString('hex')}`;
```

Comment:
Hope this is random enough to not collide

Reply:
1/18,446,744,073,709,551,615 + birthday paradox, but we won't have too many in flight at once. I think we should be fine x)